src/main/java/com/synopsys/integration/alert/component/settings/SettingsValidator.java

```diff
@@ -35,45 +31,43 @@
 import com.synopsys.integration.alert.common.enumeration.SystemMessageType;
 import com.synopsys.integration.alert.common.persistence.accessor.SystemMessageUtility;
 import com.synopsys.integration.alert.common.security.EncryptionUtility;
+import com.synopsys.integration.alert.common.system.BaseSystemValidator;
 import com.synopsys.integration.alert.component.settings.descriptor.SettingsDescriptor;
 
 @Component
-public class SettingsValidator {
+public class SettingsValidator extends BaseSystemValidator {
     private static final Logger logger = LoggerFactory.getLogger(SettingsValidator.class);
     private EncryptionUtility encryptionUtility;
-    private SystemMessageUtility systemMessageUtility;
 
     @Autowired
     public SettingsValidator(EncryptionUtility encryptionUtility, SystemMessageUtility systemMessageUtility) {
+        super(systemMessageUtility);
         this.encryptionUtility = encryptionUtility;
-        this.systemMessageUtility = systemMessageUtility;
     }
 
-    public Map<String, String> validateEncryption() {
-        systemMessageUtility.removeSystemMessagesByType(SystemMessageType.ENCRYPTION_CONFIGURATION_ERROR);
-
-        Map<String, String> fieldErrors = new HashMap<>();
+    public boolean validateEncryption() {
+        getSystemMessageUtility().removeSystemMessagesByType(SystemMessageType.ENCRYPTION_CONFIGURATION_ERROR);
+        boolean valid = true;
         if (!encryptionUtility.isInitialized()) {
             logger.error("Encryption utilities: Not Initialized");
 
-            validationCheck(SettingsDescriptor.FIELD_ERROR_ENCRYPTION_PWD, SystemMessageType.ENCRYPTION_CONFIGURATION_ERROR, !encryptionUtility.isPasswordSet())
-                .ifPresent(error -> fieldErrors.put(SettingsDescriptor.KEY_ENCRYPTION_PWD, error));
-            validationCheck(SettingsDescriptor.FIELD_ERROR_ENCRYPTION_GLOBAL_SALT, SystemMessageType.ENCRYPTION_CONFIGURATION_ERROR, !encryptionUtility.isGlobalSaltSet())
-                .ifPresent(error -> fieldErrors.put(SettingsDescriptor.KEY_ENCRYPTION_GLOBAL_SALT, error));
-            return fieldErrors;
+            boolean encryptionError = addSystemMessageIfHasError(SettingsDescriptor.FIELD_ERROR_ENCRYPTION_PWD, SystemMessageSeverity.ERROR, SystemMessageType.ENCRYPTION_CONFIGURATION_ERROR,
+                encryptionUtility.isPasswordMissing());
+            if (encryptionError) {
+                valid = false;
+                logger.error(SettingsDescriptor.FIELD_ERROR_ENCRYPTION_PWD);
+            }
+            boolean saltError = addSystemMessageIfHasError(SettingsDescriptor.FIELD_ERROR_ENCRYPTION_GLOBAL_SALT, SystemMessageSeverity.ERROR, SystemMessageType.ENCRYPTION_CONFIGURATION_ERROR,
+                encryptionUtility.isGlobalSaltMissing());
+            if (saltError) {
+                valid = false;
+                logger.error(SettingsDescriptor.FIELD_ERROR_ENCRYPTION_GLOBAL_SALT);
```

Comment:
We do this a few times, should we add an overloaded version of the method with a `logError` flag?

Reply:
The problem is we are not always logging at error level, some of them are warnings which makes it more complicated.